mycelo/contract/contracts.go

```diff
@@ -0,0 +1,261 @@
+package contract
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// CoreContracts describes the addresses for each contracts
+type CoreContracts struct {
+	BuildPath string
+
+	Libraries Libraries
+	Proxies   ContractSet
+	Contracts ContractSet
+}
+
+type ContractSet map[string]common.Address
+
+var CeloLibraries = Libraries{
+	"FixidityLib":                       common.HexToAddress("0xa001"),
+	"Proposals":                         common.HexToAddress("0xa002"),
+	"LinkedList":                        common.HexToAddress("0xa003"),
+	"SortedLinkedList":                  common.HexToAddress("0xa004"),
+	"SortedLinkedListWithMedian":        common.HexToAddress("0xa051"),
+	"AddressLinkedList":                 common.HexToAddress("0xa006"),
+	"AddressSortedLinkedList":           common.HexToAddress("0xa007"),
+	"IntegerSortedLinkedList":           common.HexToAddress("0xa008"),
+	"AddressSortedLinkedListWithMedian": common.HexToAddress("0xa009"),
+	"Signatures":                        common.HexToAddress("0xa010"),
+}
+
+func (cs ContractSet) DeployedBytecodeMappings(buildPath string, libs Libraries) (map[common.Address][]byte, error) {
+	bytecodes := make(map[common.Address][]byte)
+	for name, addr := range cs {
+		bs, err := readContractBuildFile(path.Join(buildPath, name+".json"), libs)
+		if err != nil {
+			return nil, err
+		}
+		bytecodes[addr] = bs.deployedBytecode
+	}
+	return bytecodes, nil
+}
+
+func (cs ContractSet) DeployedBytecodeFor(buildPath string, libs Libraries, name string) ([]byte, error) {
+	c, err := readContractBuildFile(path.Join(buildPath, name+".json"), libs)
+	if err != nil {
+		return nil, err
+	}
+	return c.deployedBytecode, nil
+}
+
+func (cs ContractSet) BytecodeFor(buildPath string, libs Libraries, name string) ([]byte, error) {
+	c, err := readContractBuildFile(path.Join(buildPath, name+".json"), libs)
+	if err != nil {
+		return nil, err
+	}
+	return c.bytecode, nil
+}
+
+type Libraries ContractSet
+
+func (libs Libraries) DeployedBytecodeMappings(buildPath string) (map[common.Address][]byte, error) {
+	return ContractSet(libs).DeployedBytecodeMappings(buildPath, libs)
+}
+
+func (libs Libraries) InjectLibraries(bytecode string) string {
+	for name, addr := range libs {
+		pattern := "__" + name + strings.Repeat("_", 40-4-len(name)) + "__"
+		bytecode = strings.ReplaceAll(bytecode, pattern, addr.Hex()[2:])
+	}
+	return bytecode
+}
+
+func NewCoreContracts(buildPath string) *CoreContracts {
```

Comment:
Some version of this would be helpful to have within an `Environment`. It allows to skip checking the Registry, you already have all the addresses

Reply:
Maybe have an address mapping entity that it's independent of this.. idk

Reply:
I'm not really sure what the benefit is here